### PR TITLE
[DOCU-3167] Update references to Amazon Linux 2023

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

What did you change and why?
- This PR replaces references to Amazon Linux 2022 with Amazon Linux 2023 because the OS replaced the other one completely. Anything that was supported by Amazon Linux 2022 is now supported by Amazon Linux 2023. 
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.
DOCU-3167

### Testing instructions

Netlify link: 

Testing (main pages):
- /gateway/latest/support-policy/#supported-versions
- /gateway/latest/install/linux/amazon-linux/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/gateway/install/linux/amazon-linux.md [hunk 1]
```diff
@@ -20,9 +20,51 @@ You can install {{site.base_gateway}} by downloading an installation package or
...
+Install {{site.base_gateway}} on Amazon Linux 2023 from the command line.
 
-Install {{site.base_gateway}} on Amazon Linux from the command line.
+1. Download the Kong package:
+
+{% capture download_package %}
+{% navtabs_ee codeblock %}
+{% navtab Kong Gateway %}
+```bash
+curl -Lo kong-enterprise-edition-{{page.versions.ee}}.aws.amd64.rpm "{{ site.links.download }}/gateway-3.x-amazonlinux-2023/Packages/k/kong-enterprise-edition-{{page.versions.ee}}.aws.amd64.rpm"
```
Comment (lena-larionova) [inline app/_src/gateway/install/linux/amazon-linux.md:32]: You'll need to update all of the download links in these examples to Cloudsmith links, similar to https://github.com/Kong/docs.konghq.com/pull/5841/files#diff-e21794e1a931347708305c4a77a4b38647adbfe75b1904f95455dfd508703ef3R144.

File: app/_src/gateway/install/linux/amazon-linux.md [hunk 2]
```diff
@@ -14,15 +14,57 @@ Kong is licensed under an
...
+{% capture download_package %}
+{% navtabs_ee codeblock %}
+{% navtab Kong Gateway %}
+```bash
+curl -Lo kong-enterprise-edition-{{page.versions.ee}}.rpm $(rpm --eval {{ site.links.cloudsmith }}/public/gateway-{{ page.major_minor_version }}/rpm/amzn/%{amzn}/x86_64/kong-enterprise-edition-{{page.versions.ee}}.aws.x86_64.rpm)
+```
+{% endnavtab %}
+{% navtab Kong Gateway (OSS) %}
+```bash
+curl -Lo kong-{{page.versions.ce}}.rpm $(rpm --eval {{ site.links.cloudsmith }}/public/gateway-{{ page.major_minor_version }}/rpm/amzn/%{amzn}/x86_64/kong-{{page.versions.ce}}.aws.x86_64.rpm)
```
Comment (cloudjumpercat) [inline app/_src/gateway/install/linux/amazon-linux.md:37]: @curiositycasualty Hi! Could you check these Cloudsmith package links I added for Amazon Linux 2023 to confirm if these are correct?

File: app/_src/gateway/install/linux/amazon-linux.md [hunk 3]
```diff
@@ -14,15 +14,57 @@ Kong is licensed under an
...
+
+{{ download_package | indent | replace: " </code>", "</code>" }}
+
+2. Install the package:
+
+{% capture install_package %}
+{% navtabs_ee codeblock %}
+{% navtab Kong Gateway %}
+```bash
+sudo yum install kong-enterprise-edition-{{page.versions.ee}}.aws.amd64.rpm
```
Comment (curiositycasualty) [inline app/_src/gateway/install/linux/amazon-linux.md:51]:
```suggestion
sudo yum install kong-enterprise-edition-{{page.versions.ee}}.rpm
```

File: app/_src/gateway/install/linux/amazon-linux.md [hunk 4]
```diff
@@ -63,11 +105,41 @@ sudo yum install -y kong-{{page.versions.ce}}.rpm
...
 
 {% endnavtab %}
-{% navtab YUM repository %}
+{% navtab YUM repository (AL2023) %}
+
+Install the YUM repository from the command line.
+
+1. Download the Kong APT repository:
+    ```bash
+    curl https://download.konghq.com/gateway-3.x-amazonlinux-2023/config.repo | sudo tee /etc/yum.repos.d/kong.repo
```
Comment (curiositycasualty) [inline app/_src/gateway/install/linux/amazon-linux.md:114]: This needs to match the (Amazon Linux 2) "Download the Kong YUM repository" instructions below.

File: app/_data/tables/install_options.yml [hunk 5]
```diff
@@ -32,7 +32,7 @@ features:
         enterprise: true
         support: true
         icon: /assets/images/icons/documentation/rhel.jpg
-      - name: "Amazon Linux 2"
+      - name: "Amazon Linux 2023"
```
Comment (curiositycasualty) [inline app/_data/tables/install_options.yml:35]: Don't think we want to change this. 3.4 supports 2 & 2023 according to this netlify preview.

Comment (curiositycasualty) [reply app/_data/tables/install_options.yml:35]: https://deploy-preview-5841--kongdocs.netlify.app/gateway/latest/support-policy/

File: app/_src/gateway/install/linux/amazon-linux.md [hunk 6]
```diff
@@ -14,15 +14,57 @@ Kong is licensed under an
...
+{% capture install_package %}
+{% navtabs_ee codeblock %}
+{% navtab Kong Gateway %}
+```bash
+sudo yum install kong-enterprise-edition-{{page.versions.ee}}.aws.amd64.rpm
+```
+{% endnavtab %}
+{% navtab Kong Gateway (OSS) %}
+```bash
+sudo yum install kong-{{page.versions.ce}}.aws.amd64.rpm
```
Comment (curiositycasualty) [inline app/_src/gateway/install/linux/amazon-linux.md:56]:
```suggestion
sudo yum install kong-{{page.versions.ce}}.rpm
```

**Comment by curiositycasualty** [comment]
The correct filenames are:
- `kong-3.4.0.aws.x86_64.rpm`  (CE, intel CPU architecture)
- `kong-3.4.0.aws.aarch64.rpm` (CE, arm CPU architecture)
- `kong-enterprise-edition-3.4.0.0.aws.x86_64.rpm`  (EE, intel CPU architecture)
- `kong-enterprise-edition-3.4.0.0.aws.aarch64.rpm` (EE, arm CPU architecture)

**Review by lena-larionova — APPROVED** [review]
LGTM

**Pull request merged**